### DISCUSSION TITLE
Updated and there's a couple of changes that I'd like to revert, if possible

### DISCUSSION THREAD

**Discussion by SonatasTaros** [rsxdalv/TTS-WebUI#367] [General]
The first is probably just a setting that I'm missing: in a previous version it was possible in Bark to generate 9-10 samples from one prompt. Now it seems like it's only possible to do one at a time. Is there any way to generate batches still?

I know the answer for this one is "probably not", but it's worth asking. Is there any way to restore the previous right-click behaviour in the UI? I find the most convenient way to save the samples I want to keep is to right-click and select "save as". That allows me to select where I want it saved and to rename it in the process. The right-click menu no longer offers that option, and the way that both save buttons work means that after saving you have to go find the file and move it/rename it from the default save location.

**Comment by rsxdalv** [comment]
So for the batches - you can do that in the React UI. (While Gradio
continues to improve, I still have much more potential to develop outside
of gradios restrictions)

About the right click, that seems like a problem because of gradios upgrade
changing things. But the new Gradio version includes a mini download button
that might be somewhat useful?

I see that the ultimate issue you are dealing with is that you want to only
"save" the good stuff. I intended favorites for this purpose. And what you
are asking for sounds like a "favorite" + name. Is the name always
different? For tortoise we have an option to give a name that is like a
stamp that you can change between generations.

Also, I still have the "bark legacy UI" plugin that would get generate up
to 9 samples, but it might break soon.

**Reply by SonatasTaros** [reply]
Thanks for the reply. 

WRT "React UI", I'll be honest and my knowledge of this stuff is pretty much zero. I see that there's instructions to manually install it (under a heading which says that it's not recommended to manually install things), but other than that I don't know what it is or how to access it.

Favourites doesn't really help. If you want to use multiple samples in one project then you're not only going to want them to have different names, but also names that make them easily identifiable. And if you want to create more than one project, then you're going to want to save the samples in different folders. The download button doesn't give you a choice on either of those things.

**Reply by rsxdalv** [reply]
I understand the part about managing collections and meaningful filenames. I'm thinking about save to collection button that also let's you pick a filename when you press save. Also I might be able to just add a save as button in React.

As for the React UI - it comes with the original installation. When the server starts it also starts the React UI assuming nothing is completely broken. Then when you see the Gradio (default) UI, there should be a blue link at the top saying React UI. It should be https://localhost:3000/ by default.

**Reply by SonatasTaros** [reply]
Thanks, I don't know how I missed that link. The instillation I have (the most recent) actually opens that page first in the browser, but it gets "unable to connect". Same happens once everything is fully loaded and I refresh the page, or visit the link at the top of the gradio page.

**Reply by rsxdalv** [reply]
I see, in that case it might be showing some errors in the console when it starts up. If you have the time, please send me a copy of logs from the console when you start the app. Also the _bark legacy plugin_ was updated so that it continues to function a bit longer.

**Reply by rsxdalv** [reply]
By the way, I checked and the save as will work in the React UI for the foreseeable future, by right clicking on the download button.

**Reply by rsxdalv** [reply]
Final update - I noticed that even with the new Gradio, if you right click on the download button you can do save as:

![image](https://github.com/user-attachments/assets/3f43254e-36ad-45fa-80ea-e840cf5d81ed)
